numpy/lib/utils.py

```diff
@@ -104,6 +104,26 @@ def newfunc(*args,**kwds):
         if doc is None:
             doc = depdoc
         else:
+            # We want to indent the original first line by as much as
+            # inspect.cleandoc() will dedent second and following lines.
+            lines = doc.expandtabs().split('\n')
+            margin = sys.maxsize
+            for line in lines[1:]:
+                content = len(line.lstrip())
+                if content:
+                    indent = len(line) - content
+                    margin = min(margin, indent)
+            if margin == sys.maxsize:
+                margin = 0
+            if lines.pop(0).lstrip():
+                doc = margin * ' ' + doc
+            else:
+                # Remove the same leading "blank" lines as cleandoc() would
+                # without actually stripping any line or expanding tabs.
+                start = doc.index('\n') + 1
+                while lines and len(lines.pop(0)) <= margin:
+                    start = doc.index('\n', start) + 1
```

Comment:
This loop would be less confusing as:
```
start = 0
length = 0
while length <= margin:
    end = doc.index('\n', start)
    length = end - start
    start = end + 1
```

Right now, the same state is in both `lines` and `doc`, which is pretty confusing

Reply:
Rewritten the loop to avoid an exception for empty docstrings and hopefully make it more readable (another edge case is a whitespace-only first line, that should also be removed).

Reply:
Noticed that, thanks!